PiEventCallCodec Bson codec can't serialise MetadataAtomicProcesses

PiEventCallCodec casts to an AtomicProcess internally, causing crashes when passed MetadataAtomicProcesses. 

`org.apache.kafka.common.errors.SerializationException: Error deserializing key/value for partition Result-0 at offset 0. If needed, please seek past the record to continue consumption.
Caused by: java.lang.ClassCastException: workflow.monitor.instances.InitPressingInstance cannot be cast to com.workflowfm.pew.AtomicProcess
        at com.workflowfm.pew.mongodb.bson.events.PiEventCallCodec.decodeBody(PiEventCallCodec.scala:51)
        at com.workflowfm.pew.mongodb.bson.events.PiEventCallCodec.decodeBody(PiEventCallCodec.scala:9)
        at com.workflowfm.pew.mongodb.bson.auto.ClassCodec.decode(AutoCodecTypes.scala:34)`

--- a/src/com/workflowfm/pew/mongodb/bson/events/PiEventCallCodec.scala
+++ b/src/com/workflowfm/pew/mongodb/bson/events/PiEventCallCodec.scala
@@ -47,9 +47,9 @@ class PiEventCallCodec[T](
 
     val ref: Int = reader.readInt32( callRefN )
 
-    val proc: AtomicProcess
+    val proc: MetadataAtomicProcess
       = ctx.decodeWithChildContext( procCodec, reader )
-        .asInstanceOf[AtomicProcess]
+        .asInstanceOf[MetadataAtomicProcess]
 
     val args: Seq[PiObject]
       = readArray( reader, argsN ) { () =>

